tuxemon/item/effects/learn_mm.py

```diff
@@ -0,0 +1,45 @@
+from __future__ import annotations
+
+import random
+from dataclasses import dataclass
+
+from tuxemon.db import ElementType, db
+from tuxemon.item.itemeffect import ItemEffect, ItemEffectResult
+from tuxemon.monster import Monster
+from tuxemon.technique.technique import Technique
+
+
+class LearnMmEffectResult(ItemEffectResult):
+    pass
+
+
+@dataclass
+class LearnMmEffect(ItemEffect):
+    """This effect teaches the target a random type technique."""
+
+    name = "learn_mm"
+    element: ElementType
+
+    def apply(self, target: Monster) -> LearnMmEffectResult:
+        techs = list(db.database["technique"])
+        # type moves
+        filters = []
+        for mov in techs:
+            results = db.lookup(mov, table="technique")
+            if self.element in results.types:
+                filters.append(results.slug)
+        # monster moves
+        moves = []
+        for tech in target.moves:
+            moves.append(tech.slug)
+        # remove monster moves from type moves
+        set1 = set(filters)
+        set2 = set(moves)
+        res = list(set1 - set2)
+        # add a random move from what remains
+        if len(res) > 0:
```

Comment:
`        if res:`

Reply:
It is generally best to do 

`        if res:`
than
`        if len(res) > 0:`

Mainly because it's more concise.